### PR TITLE
experimentally enable vcpu hotplug and virtio-mem on arm64 in kernel part

### PR DESCRIPTION
**Pull request by jongwu**
VCPU hotplug is the key feature for kata to dynamically depoly resources. Unfortunately, up to now, this feature is yet on the road and we can't foresee when will come to reality. Thanks to the initial work from Salli from Huawei, we can try vcpu hotplug on arm64. I port their kernel patches to 5.15 and do some test and bug fix. Even this is far to be a product, I want to contribute it here to be a ‘Reference Implementation’. Thus user can try vcpu hotplug or do more testes and improve it.

As this is far from perfect and has not been accept by upstream, I put all of the patches to experimental area.
Also I twist the kata to let vcpu hotplug work on arm64.

### CODE REVIEW

**Comment by jongwu** [comment]
@jodh-intel @fidencio @devimc

File: tools/packaging/kernel/build-kernel.sh
Comment (jodh-intel) [inline tools/packaging/kernel/build-kernel.sh:501]: Since this isn't just "experimental", this makes it an "architecture specific experimental experimental kernel" right? This could get confusing :smile:

**Review by jodh-intel — DISMISSED** [review]
Thanks @jongwu.

lgtm but let's get @jcvenegas to tal too...

File: tools/packaging/kernel/build-kernel.sh
Comment (jongwu) [reply tools/packaging/kernel/build-kernel.sh:501]: Yeah, this is arch-specific experimental kernel. Should we add a "arm-experimental" opt to "build-kernel.sh" to eliminate this confusion?

Comment (jodh-intel) [reply tools/packaging/kernel/build-kernel.sh:501]: It might be better if we added a new generic option that would only be used if `build_type=experimental"`. That would be a more generic solution and might be clearer. But I'm not sure what that new option would be called ("architecture specific experimental mode"?) But if we can find a name, if that option was specified, the script would use the version defined by `assets.${arch}-kernel-experimental.version` in the versions db.

Comment (jongwu) [reply tools/packaging/kernel/build-kernel.sh:501]: What about adding a new option "-E" for build_kernel.sh and the corresponding build_type would be "arch-experimental". Also add "assets.${arch}-kernel-experimental.version" to versions db. Thus, both of "-a" and "-E" in build_kernel.sh specify the "${arch}-kernel-experimental.version".

Comment (jodh-intel) [reply tools/packaging/kernel/build-kernel.sh:501]: @jongwu - could you tal at https://github.com/kata-containers/ci/pull/406 as this relates to experimental kernels too.

File: tools/packaging/kernel/build-kernel.sh
Comment (Jakob-Naucke) [inline tools/packaging/kernel/build-kernel.sh:90]: Nit: Can you indent with spaces here? LGTM otherwise.

Comment (jongwu) [reply tools/packaging/kernel/build-kernel.sh]: thanks @Jakob-Naucke, done.

**Comment by fidencio** [comment]
A head's up that may affect your PR.

In the Architecture Committee meeting from January 25th, 2022, the Architecture Committee has agreed on using the "Dismiss stale pull request approvals when new commits are pushed" configuration from GitHub.  It basically means that if your PR has been rebased or updated, the approvals given will be erased.

In order to minimize traumas due to the new approach, please, consider adding a note on the changes done before the rebase / force-push, and also pinging the reviewers for a subsequent round of reviews.

Thanks for your understanding!

Related issue: https://github.com/kata-containers/community/issues/249

File: tools/packaging/kernel/build-kernel.sh
Comment (Jakob-Naucke) [reply tools/packaging/kernel/build-kernel.sh]: I'm sorry, but it's indented too far.

Comment (jongwu) [reply tools/packaging/kernel/build-kernel.sh]: what about this?

Comment (Jakob-Naucke) [reply tools/packaging/kernel/build-kernel.sh]: Thanks @jongwu, I'll TAL at the rest again... (btw, I said "indent with spaces", that was incorrect, we're indenting with tabs here. I meant "align with spaces". Sorry!)

File: tools/packaging/kernel/build-kernel.sh
Comment (Jakob-Naucke) [reply tools/packaging/kernel/build-kernel.sh:508]: I'd be fine with `arch-experimental`, but it would be good to come to a consensus here.

**Review by Jakob-Naucke — APPROVED** [review]
LGTM (although I did not review all the QEMU patches).

File: tools/packaging/kernel/build-kernel.sh
Comment (jongwu) [reply tools/packaging/kernel/build-kernel.sh:508]: thanks @Jakob-Naucke -, any idea? @devimc @jodh-intel

**Comment by fidencio** [comment]
@jcvenegas, @gkurz, mind to also take a look on this one?

**Review by jodh-intel — APPROVED** [review]
Thanks @jongwu.

**Comment by jongwu** [comment]
Hello @fidencio, this PR needs rebase. So, it needs approve again after re-pushed, is it right?

**Comment by Jakob-Naucke** [comment]
@jongwu, yes we recently enabled the requirement for reapprovals after updates, and that does include rebases.

**Comment by jongwu** [comment]
@bergwolf -, let me rebase it later.

**Pull request merged**
